app/assets/javascripts/editor.js

```diff
@@ -988,29 +1004,46 @@ $(function() {
     });
   };
 
+  var showWebsocketError = function() {
+    $.flash.danger({
+      text: $('#flash').data('message-failure')
+    });
+  }
+
   var showWorkspaceTab = function(event) {
     event.preventDefault();
-    showTab(1);
+    showTab(0);
   };
 
   var stopCode = function(event) {
     event.preventDefault();
     if ($('#stop').is(':visible')) {
-      var jqxhr = ajax({
```

Comment:
This might stilll be needed for non websocket calls like assess, i suggest have a two function or one that takes an param execution_type